plugins/lttng-utils/debug-info/fdcache.h

```diff
@@ -0,0 +1,51 @@
+#ifndef BABELTRACE_PLUGIN_DEBUG_INFO_FD_CACHE_H
+#define BABELTRACE_PLUGIN_DEBUG_INFO_FD_CACHE_H
+/*
+ * fdcache.h
+ *
+ * Babeltrace - File descriptor cache for debug-info
+ *
+ * Copyright 2019 Francis Deslauriers <francis.deslauriers@efficios.com>
+ *
+ * Author: Francis Deslauriers <francis.deslauriers@efficios.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <inttypes.h>
+
+struct fd_handle {
+	int fd;
+};
+
+struct fd_cache;
+
+BT_HIDDEN
+struct fd_cache *fd_cache_create(void);
```

Comment:
No need for dynamic allocation here I guess? Just have `bt_fd_cache_init()` and `bt_fd_cache_fini()` which do not dynamically allocate and have a simple `struct bt_fd_cache fd_cache` in `struct debug_info_msg_iter`.